src/test/java/org/greenbuttonalliance/gbaresourceserver/customer/repository/CustomerRepositoryTest.java

```diff
@@ -0,0 +1,151 @@
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+@DataJpaTest(showSql = false)
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+public class CustomerRepositoryTest {
+	@Autowired
+	private CustomerRepository customerRepository;
+
+	// for testing findById
+	private static final String PRESENT_PUC = "foo";
+	private static final String NOT_PRESENT_PUC = "bar";
+
+	@BeforeEach
+	public void initTestData() {
+		customerRepository.deleteAllInBatch();
+		customerRepository.saveAll(buildTestData());
+	}
+
+	@Test
+	public void findByPresentId_returnsMatching() {
+		UUID presentUuid = UuidCreator.getNameBasedSha1(UuidCreator.NAMESPACE_URL, PRESENT_PUC);
+		UUID foundUuid = customerRepository.findById(presentUuid).map(Customer::getUuid).orElse(null);
+
+		Assertions.assertEquals(
+			presentUuid,
+			foundUuid,
+			() -> String.format("findById with %s returns entity with ID %s", presentUuid, foundUuid)
+		);
+	}
+
+	@Test
+	public void findByNotPresentId_returnsEmpty() {
+		UUID notPresentUuid = UuidCreator.getNameBasedSha1(UuidCreator.NAMESPACE_URL, NOT_PRESENT_PUC);
+		Optional<Customer> customer = customerRepository.findById(notPresentUuid);
+
+		Assertions.assertTrue(
+			customer.isEmpty(),
+			() -> String.format("findById with %s returns entity with ID %s", notPresentUuid, customer.map(Customer::getUuid).orElse(null))
+		);
+	}
+
+	@Test
+	public void findAll_returnsAll() {
+		int findByAllSize = customerRepository.findAll().size();
+		int testDataSize = buildTestData().size();
+
+		Assertions.assertEquals(
+			findByAllSize,
+			testDataSize,
+			() -> String.format("findByAll size of %s does not match test data size of %s", findByAllSize, testDataSize)
+		);
+	}
+
+	@Test
+	public void entityMappings_areNotNull() {
+		Customer fullyMappedCustomer = customerRepository.findById(UuidCreator.getNameBasedSha1(UuidCreator.NAMESPACE_URL, PRESENT_PUC)).orElse(null);
+		Assumptions.assumeTrue(fullyMappedCustomer != null);
+
+		Function<Customer, Optional<Organisation>> customerToOrganisation = c -> Optional.ofNullable(c.getOrganisation());
+		Function<Customer, Optional<Status>> customerToStatus = c -> Optional.ofNullable(c.getStatus());
+		Function<Customer, Optional<Priority>> customerToPriority = c -> Optional.ofNullable(c.getPriority());
+
+		Assertions.assertAll(
+			"Entity mapping failures for customer " + fullyMappedCustomer.getUuid(),
+			Stream.of(customerToOrganisation,
+					customerToStatus,
+					customerToPriority)
+				.map(mappingFunc ->
+					() -> Assertions.assertTrue(mappingFunc.apply(fullyMappedCustomer).isPresent()))
+		);
+	}
+
+	private static List<Customer> buildTestData() {
+		List<Customer> customers = Arrays.asList(
+			Customer.builder()
+				.organisation(Organisation.builder()
+					.organisationName("foo")
+					.build())
```

Comment:
This `Organisation` probably looks a little light, but the full `Organisation` mappings are already tested in the `CustomerAccountRepositoryTest`, and I didn't want to unnecessarily repeat that data and those tests here. Same goes for the `Organisation` in the `ServiceSupplierTest`.